frontend/src/v5/store/containers/containers.types.ts

```diff
@@ -41,3 +43,18 @@ export interface IContainer {
 	isFavourite: boolean;
 	role: string;
 }
+
+export interface FavouritePayload {
+	teamspace: string;
+	projectId: string;
+	containerId: IContainer['_id'];
+}
+
+export interface IContainersActionCreators {
+	setFilterQuery: (query: string) => ExtendedAction<{query: string}, 'setFilterQuery'>
+	addFavourite: (teamspace: string, projectId: string, containerId: string) => ExtendedAction<FavouritePayload, 'addFavourite'>
+	removeFavourite: (teamspace: string, projectId: string, containerId: string) => ExtendedAction<FavouritePayload, 'removeFavourite'>
+	toggleFavouriteSuccess: (containerId: string) => ExtendedAction<{containerId: string}, 'toggleFavouriteSuccess'>
+}
```

Comment:
TBH I dont think ExtendedAction<{query: string}, 'setFilterQuery'> is too readable. I think is more readable to have something like
```export type FilterQueryAction = Action<'SET_FILTER_QUERY'> & {query: string};```
for all actions instead of them getting inferred and accessed via 'functionName'